inspirehep/records/api/literature.py

```diff
@@ -143,3 +152,90 @@ def _dump_for_es(self):
         serialized_data = super()._dump_for_es()
         serialized_data["_ui_display"] = json.dumps(self.get_ui_data())
         return serialized_data
+
+
+def harvest_article(id_type: str, identifier: str) -> dict:
+    """Harvest the article corresponding to `identifier`.
+
+    This function queries an endpoint on `inspire-next` to harvest a record
+    either from arXiv or Crossref, based on the given id_type and identifier.
+
+    Args:
+        id_type(str): it can be `arxiv` or `doi`.
+        identifier(str): normalized arxiv/doi identifier.
+
+    Returns:
+        dict: the article harvested in JSON format, if any.
+
+    Raises:
+        UnknownIdentifierError: if `id_type` is neither "arxiv" or "doi".
+    """
+    if id_type not in ("arxiv", "doi"):
+        raise UnknownIdentifierError(identifier)
+
+    inspire_next_url = current_app.config["INSPIRE_NEXT_URL"]
+    endpoint = (
+        current_app.config["INSPIRE_NEXT_IMPORT_ARXIV_ENDPOINT"]
+        if id_type == "arxiv"
+        else current_app.config["INSPIRE_NEXT_IMPORT_DOI_ENDPOINT"]
+    )
+
+    identifier_param = requests.utils.quote(identifier, safe="")
+
+    url = f"{inspire_next_url}{endpoint}/{identifier_param}"
+
+    headers = {
+        "content-type": "application/json",
+        "Authorization": f"Bearer {current_app.config['AUTHENTICATION_TOKEN']}",
+    }
+
+    response = requests.post(url=url, headers=headers)
+
+    if response.status_code == 200:
+        return response.json()
+
+
+def import_article(identifier: str) -> dict:
+    """Import a new article from arXiv or Crossref based on the identifier.
+
+    This function attempts to parse  and normalize the identifier as a valid
+    arXiv id or DOI. If the identifier is valid and there is no record in
+    Inspire matching the ID, it requests the article through inspire-next's
+    APIs.
+
+    Args:
+        identifier(str): the ID of the record to import
+
+    Returns:
+        dict: the serialized article
+
+    Raises:
+        UnknownIdentifierError: if the identifier is not an arXiv id or a DOI.
+        ExistingArticleError: if the record is already in Inspire.
+        ImportRecordError: if the article is not found.
+    """
+
+    if is_arxiv(identifier):
+        pid_type = "arxiv"
+        pid_value = normalize_arxiv(identifier)
+
+    elif is_doi(identifier):
+        pid_type = "doi"
+        pid_value = normalize_doi(identifier)
+
+    else:
+        raise UnknownIdentifierError(identifier)
+
+    pid = PersistentIdentifier.query.filter_by(
```

Comment:
I feel like this will be done many times for other pids, it would be good to have  `def exists(pid_type, pid_value)` in `PersistentIdentifier` class.

Reply:
I agree with that, but this shouldn't be added to this task, we should create a new task for this.

Reply:
I mean it's a few lines function at the end, separate commit would work OK, IMO